state-sync/aptos-data-client/src/peer_states.rs

```diff
@@ -435,6 +462,37 @@ pub fn get_bucket_id_for_peer(peer: PeerNetworkId) -> u8 {
     peer_id_bytes[0] % NUM_PEER_BUCKETS_FOR_METRICS
 }
 
+/// Updates the metrics for the number of ignored peers
+fn update_peer_ignored_metrics(peer_to_state: Arc<DashMap<PeerNetworkId, PeerState>>) {
+    // Collect the ignored peer counts by network
+    let mut ignored_peer_counts_by_network: BTreeMap<NetworkId, u64> = BTreeMap::new();
+    for peer_state_entry in peer_to_state.iter() {
+        // Get the peer and state
+        let peer = *peer_state_entry.key();
```

Comment:
move most of this inside `if peer_state.is_ignored()` ?

Reply:
`is_ignored()` is called super often, and I only want to update the metrics infrequently. Also, this metrics need network ID information, so moving things across would be messier 😄

Reply:
I mean within this block. Most of this block can be moved to within the `if peer_state.is_ignore(){}` block at the end of it.

Reply:
Aah, gotcha. I did it this way so that we can always be sure that `ignored_peer_counts_by_network` contains entries for networks with `0` ignored peers. That way, when we update the metrics, the graphs show `0` (instead of just nothing). Otherwise, in most cases, the metric is not updated.